File: Local API/C#.NET/Messages/Delete/Delete.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Delete
{
  class Delete
  {
    static void Main(string[] args)
    {
      AXMMCFGLib.Database objDatabase = new AXMMCFGLib.Database();
      AXMMCFGLib.Constants objConstants = new AXMMCFGLib.Constants();
      string strFilter = "";

      Console.WriteLine("API Module: {0}\nAPI Build: {1}\n", objDatabase.Module, objDatabase.Build);

      var objMessageDB = objDatabase.GetMessageDB();
      Console.WriteLine("Open, result: {0} ({1})", objDatabase.LastError, objDatabase.GetErrorDescription(objDatabase.LastError));
      if (objMessageDB.LastError != 0)
        return;

      strFilter = "ID > 2 And ID < 4 And Direction='OUT'";
      objMessageDB.Delete("", strFilter);
      Console.WriteLine("Delete [{0}], result: {1} ({2})", strFilter, objMessageDB.LastError, objMessageDB.GetErrorDescription(objMessageDB.LastError));

      objMessageDB.NotifyNonStatusUpdate(); // 'non-status-update' is a trivial update; only the UI's will update.

      Console.WriteLine("Ready.");
    }
  }
}

File: Local API/VB.NET/Messages/Delete/Delete.vb
Module Delete

  Sub Main()
    Dim objDatabase As AXMMCFGLib.Database = New AXMMCFGLib.Database()
    Dim objConstants As AXMMCFGLib.Constants = New AXMMCFGLib.Constants()
    Dim strFilter As String = ""

    Console.WriteLine("API Module: {0}" & vbLf & "API Build: {1}" & vbLf, objDatabase.[Module], objDatabase.Build)
    Dim objMessageDB = objDatabase.GetMessageDB()

    Console.WriteLine("Open, result: {0} ({1})", objDatabase.LastError, objDatabase.GetErrorDescription(objDatabase.LastError))
    If objMessageDB.LastError <> 0 Then Return

    strFilter = "ID > 2 And ID < 4 And Direction='OUT'"
    objMessageDB.Delete("", strFilter)
    Console.WriteLine("Delete [{0}], result: {1} ({2})", strFilter, objMessageDB.LastError, objMessageDB.GetErrorDescription(objMessageDB.LastError))

    objMessageDB.NotifyNonStatusUpdate()

    Console.WriteLine("Ready.")
  End Sub

End Module
